btr-api/src/btr_api/resources/submission.py

```diff
@@ -40,18 +40,132 @@
+        return payload
+    # otherwise PUBLIC or STAFF
+    redactionToUse = REDACT_RULES[role]
+    for person in payload["payload"]["personStatements"]:
+        for name in person["names"]:
+            if name["type"] == "alternative":
+                name["fullName"] = redactField(name["fullName"], redactionToUse.get("prefName"))
+
+        person["email"] = redactField(person["email"], redactionToUse.get("email"))
+        person["phoneNumber"]["number"] = redactField(person["phoneNumber"]["number"], redactionToUse.get("phone"))
+        for address in person["addresses"]:
+            address["postalCode"] = redactField(address["postalCode"], redactionToUse.get("postal"))
+            address["street"] = redactField(address["street"], redactionToUse.get("street"))
+            address["streetAdditional"] = redactField(
+                address["streetAdditional"], redactionToUse.get("streetAdditional")
+            )
+            address["locationDescription"] = redactField(
+                address["locationDescription"], redactionToUse.get("locationDescription")
+            )
+
+        person["placeOfResidence"]["postalCode"] = redactField(
+            person["placeOfResidence"]["postalCode"], redactionToUse.get("postal")
+        )
+        person["placeOfResidence"]["street"] = redactField(
+            person["placeOfResidence"]["street"], redactionToUse.get("street")
+        )
+        person["placeOfResidence"]["streetAdditional"] = redactField(
+            person["placeOfResidence"]["streetAdditional"], redactionToUse.get("streetAdditional")
+        )
+        person["placeOfResidence"]["locationDescription"] = redactField(
+            person["placeOfResidence"]["locationDescription"], redactionToUse.get("locationDescription")
+        )
+        person["birthDate"] = redactField(person["birthDate"], redactionToUse.get("birthDate"))
+        for identifier in person["identifiers"]:
+            identifier["id"] = redactField(identifier["id"], redactionToUse.get("identifiers"))
+    return jsonify(payload)
+
+
+def redactField(field, redactType):
+    if redactType == REDACT_MONONYM:
+        return field[0:1] + "***"
+    elif redactType == REDACT_MONONYM_FN:
+        rv = ""
+        for word in field.split():
+            if not (rv == ""):
+                rv += " "
+            rv += word[0:1] + "***"
+        return rv
+    elif redactType == REDACT_EMAIL:
+        return field.split("@")[0][0:1] + "***" + "@***." + field.split(".")[-1]
+    elif redactType == REDACT_PHONE:
+        return field[0:-7] + "***"
+    elif redactType == REDACT_FULL:
+        return "***"
+    # This is a space instead of blank because if it's empty the ui shows undefined in some spots
+    elif redactType == REDACT_EMPTY:
+        return " "
+    elif redactType == REDACT_DATE:
+        return field.split("-")[0]
+    elif redactType == REDACT_IDENTIFIER:
+        return "*** **" + field[6:]
+
+    return field
```

Comment:
pls move the redact code into a util or the submission service

Reply:
+1 Yes, please put it in the util, I think the best is if service gets all data as is, and the data can be obfuscated wherever it needs to be.